### PR TITLE
[sival,tests] Allow excessive csrng irqs in plic_all_irqs_test

### PR DESCRIPTION
**Pull request by matutem**
Carefully allow these excessive irqs for the correct portion of
the tests, addressing https://github.com/lowRISC/opentitan/issues/20747 so this test passes.

Add a TODO against the issue above so the code is adjusted
depending on how the issue is addressed.

Enable these tests in fpga_cw310_sival_rom_ext.

### CODE REVIEW

**Comment by msfschaffner** [comment]
Should we fix this in the ROM_EXT instead?
Or maybe add a comment in the code that the spurious IRQs are likely due to ROM_EXT behavior?

File: util/topgen/templates/plic_all_irqs_test.c.tpl [hunk 1]
```diff
@@ -98,45 +110,70 @@ void ottf_external_isr(uint32_t *exc_info) {
 
   top_${top["name"]}_plic_peripheral_t peripheral = (top_${top["name"]}_plic_peripheral_t)
       top_${top["name"]}_plic_interrupt_for_peripheral[plic_irq_id];
-  CHECK(peripheral == peripheral_expected,
-        "Interrupt from incorrect peripheral: exp = %d, obs = %d",
-        peripheral_expected, peripheral);
+  if (allow_csrng_irq && kBootStage == kBootStageOwner &&
+      peripheral != peripheral_expected &&
+      peripheral == kTopEarlgreyPlicPeripheralCsrng) {
```
Comment (msfschaffner) [inline util/topgen/templates/plic_all_irqs_test.c.tpl:115]: Should we refer to #20747 at this point?

Comment (matutem) [reply util/topgen/templates/plic_all_irqs_test.c.tpl:118]: I added references at other points. A reference here will not hurt. Done.

**Review by moidx — APPROVED** [review]

**Comment by moidx** [comment]
We should consider writing an entropy complex test to tweak the EDN settings and monitors this interrupt. I am not sure which EDN endpoint is causing the interrupt to trigger so often. Maybe enabling the EDN `edn_cmd_req_done` interrupt may help to root cause the issue?

**Comment by matutem** [comment]
The failure is unrelated, so this is merged.

**Pull request merged**